moveit_core/robot_state/test/attached_body_test.cpp

```diff
@@ -0,0 +1,133 @@
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the Willow Garage nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Tom Noble */
+
+#include <geometric_shapes/shapes.h>
+#include <moveit/robot_state/attached_body.hpp>
+#include <moveit/robot_model/robot_model.hpp>
+#include <moveit/robot_state/robot_state.hpp>
+#include <urdf_parser/urdf_parser.h>
+#include <gtest/gtest.h>
+
+class SingleLinkRobot : public ::testing::Test
+{
+public:
+  SingleLinkRobot()
+  {
+    static const std::string URDF_XML = R"(
+      <?xml version="1.0" ?>
+      <robot name="single_link_robot">
+      <link name="link">
+        <inertial>
+          <mass value="2.81"/>
+          <origin rpy="0 0 0" xyz="0.0 0.0 .0"/>
+          <inertia ixx="0.1" ixy="-0.2" ixz="0.5" iyy="-.09" iyz="1" izz="0.101"/>
+        </inertial>
+        <collision name="my_collision">
+          <origin rpy="0 0 0" xyz="0 0 0"/>
+          <geometry>
+            <box size="1 2 1" />
+          </geometry>
+        </collision>
+        <visual>
+          <origin rpy="0 0 0" xyz="0.0 0 0"/>
+          <geometry>
+            <box size="1 2 1" />
+          </geometry>
+        </visual>
+      </link>
+      </robot>
+    )";
+
+    static const std::string SRDF_XML = R"xml(
+      <?xml version="1.0" ?>
+      <robot name="single_link_robot">
+      </robot>
+      )xml";
+
+    urdf::ModelInterfaceSharedPtr urdf_model = urdf::parseURDF(URDF_XML);
+    auto srdf_model = std::make_shared<srdf::Model>();
+    srdf_model->initString(*urdf_model, SRDF_XML);
+    robot_model_ = std::make_shared<moveit::core::RobotModel>(urdf_model, srdf_model);
+  }
+
+protected:
+  moveit::core::RobotModelConstPtr robot_model_;
+};
+
+class SingleAttachedBody : public SingleLinkRobot
+{
+public:
+  SingleAttachedBody()
```

Comment:
Do you want this to be a subclass of the above test fixture, or just have it be a single test?

like, I can imagine all this code could be part of the body of `TEST_F(SingleAttachedBody, RootBodyHasCorrecAttachedLink)`, and any repetitive code as you add more tests can be a helper function / method of the original fixture.

This would
* Save you the extra layer of inheritance
* Make the test not just be a one liner